src/net.cpp

```diff
@@ -1681,8 +1683,10 @@ void CConnman::ThreadOpenConnections()
             if (nANow - addr.nLastTry < 600 && nTries < 30)
                 continue;
 
-            // only consider nodes missing relevant services after 40 failed attempts and only if less than half the outbound are up.
-            if ((addr.nServices & nRelevantServices) != nRelevantServices && (nTries < 40 || nOutbound >= (nMaxOutbound >> 1)))
+            // only consider nodes missing relevant services we have at least 4 already,
+            // and at least 6 once SegWit has activated
+            if ((addr.nServices & nRelevantServices) != nRelevantServices && (nTries < 40 &&
+                    nRelevant < 5) && !fFeeler)
```

Comment:
I felt a minimum of 4 was a better criteria than half of the default 8 as if someone reduced maxOutbound then I think fewer than 4 is too few.